### DISCUSSION TITLE
Beating Six Flags Magic Mountains in 1.56 seconds by selling everything bug?

### DISCUSSION THREAD

**Discussion by ageekhere** [OpenRCT2/OpenRCT2#17935] [Feature requests]
Beating Six Flags Magic Mountains in 1.56 seconds by selling everything bug?
 
https://www.youtube.com/watch?v=V2mscAQFSF0&ab_channel=MarcelVos

I would have thought that by selling everything in the park that it would drop the park value to nil, as the missing asks for a park value of at least 700,000 and not a company value of at least 700,000.

Is this a bug with park value maybe not updating quick enough so the mission passes?

**Comment by fidwell** [comment]
He doesn't sell everything, just two rides.

**Reply by ageekhere** [reply]
Ah i see the total park value is 804,261 so 100,000 worth of rides can be sold off as you only need 700,000 park value to complete the mission and you only have a 50,000 load with 10,000 cash so only 40,000 worth of rides needs to be sold off to complete the mission.

Hmm so not a game bug but rather a mission requirement oversight,
Maybe it should be have a park value over 850,000 instead.

**Reply by James103** [reply]
> Maybe it should be have a park value over 850,000 instead

This could be done in a plugin as follows:
* When loading a scenario, check if the name is "Six Flags Magic Mountain" and the goal is "Repay loan and park value" with the park value requirement being "700,000". If so, then change that "700,000" to "850,000" (or some other value that can't be reached by selling a few rides quickly).

Alternatively or in addition to the above,
* Make all the rides a couple of years old, thereby reducing both park value and their sell value.
* Increase starting loan to 75–120k (or some other value that can't be reached by selling a few rides quickly) without increasing starting cash.

**Reply by ageekhere** [reply]
I wonder if Mission requirement oversights should be fix for the base game or in plugins.

Does anyone know how many other missions have issues like this one?

**Reply by fidwell** [reply]
Many of the expansion scenarios have poor goals. (In Heide Park, you can win without doing anything at all.) But it is not the goal of OpenRCT2 to "fix" these.

**Reply by ocalhoun6** [reply]
Yeah, got to agree with that. And a _lot_ of the scenarios are ridiculously easy to complete if you know how to exploit the game mechanics -- such as just building a few ultra-extreme intensity coasters in any park value scenario, or spamming microcoasters in any scenario with a guest goal.

I don't really think that this is a _problem_ that needs to be fixed.

It's okay that people who are very good at the game will be able to beat a lot of scenarios very easily. By the point that players are this advanced, they're probably not looking to scenarios for a real challenge anyway. Much more likely to be interested in creative play, multiplayer, or competitive speedrunning or something. (And I definitely don't think it's our job to arbitrarily decide that speedrunning should be more difficult.)

I think that -- in general -- we should keep modifications to original scenarios to a minimum, only fixing things that are very obviously bugs, like the individual squares of un-owned land, or the one with a manta ray coaster that crashes when a bug with that coaster is fixed.

**Comment by spacek531** [comment]
1.56 seconds is in realtime not gametime. He uses hotkeys to open windows and change the simulation speed so that 1 month takes 1 second.
